sphinxcontrib/autodoc_pydantic/directives/autodocumenters.py

```diff
@@ -324,10 +334,43 @@ def add_collapsable_schema(self):
 
         schema_rest = self._convert_json_schema_to_rest(schema)
         source_name = self.get_sourcename()
-
         for line in schema_rest:
             self.add_line(line, source_name)
 
+    def add_erdantic_figure(self, collapsed: bool):
+        """Adds an erdantic entity relation diagram to the doc of an
+        pydantic model.
+
+        """
+        source_name = self.get_sourcename()
+        if erd is None:
+            error_msg = 'erdantic is not installed, you need to install ' \
+                'it before creating an Entity Relationship Diagram for ' \
+                'f{self.fullname}. See ' \
+                'https://autodoc-pydantic.readthedocs.io/' \
+                'en/stable/users/installation.html'
+            raise RuntimeError(error_msg)
+
+        # Graphviz [DOT language](https://graphviz.org/doc/info/lang.html)
+        figure_dot = erd.to_dot(self.object)
+        lines = [
+            ".. graphviz::",
+            "",
+        ]
+        lines += ['   ' + line
+                  for line in figure_dot.replace('\t', '   ').split('\n')]
+        lines.append("")
+
+        if collapsed:
+            lines = "\n".join(lines)
+            lines = TPL_COLLAPSE.format(
+                lines=lines,
+                summary="Show Entity Relationship Diagram",
+                details_class="autodoc_pydantic_collapsable_erd"
+            ).split("\n")
```

Comment:
This block appears twice since we have the second feature that uses collapsible blocks (`autodoc_pydantic_collapsable_json`). To prevent code duplication, I suggest we use a function like this located in `sphinxcontrib/autodoc_pydantic/directives/templates.py`:
```python
def to_collapsable(lines: list[str], title, css_class) -> list[str]:
    """Place given lines into a collapsable HTML block.
    
    Parameters
    ----------
    lines: list
        The actual content, that should be placed into a collapsable block.
    title: str
        The name of the collapsable block title.
    css_class: str
        Name of the css class for the collapsable block.
        
    """
    
    lines = "\n".join(lines)
    lines = TPL_COLLAPSE.format(
        lines=lines,
        summary=title,
        details_class=css_class
    )
    return lines.split("\n")
```

Then, we could change the marked code to:

```python
if collapsed:
  lines = to_collapsable(
      lines=lines,
      title="Show Entity Relationship Diagram",
      css_class="autodoc_pydantic_collapsable_erd"
  )
```

Reply:
fixed in 67475f0